src/modules/quick_calculated_module.f90

```diff
@@ -318,13 +318,17 @@ subroutine reallocate_quick_qm_struct(self,ierr)
      integer, intent(inout) :: ierr
      integer :: current_size     
 
-     if(quick_molspec%nextatom .gt. 0) then
-       if(allocated(self%ptchg_gradient)) current_size= size(self%ptchg_gradient)
-       if(current_size /= quick_molspec%nextatom*3) then
+     if (quick_molspec%nextatom > 0) then
+       if (allocated(self%ptchg_gradient)) then
+         current_size = size(self%ptchg_gradient)
+       else
+         current_size = 0
+       endif
+       if (current_size < 3*quick_molspec%nextatom) then
```

Comment:
This guarantees that we have sufficient memory. But it could lead to issues because we may assume somewhere else in the code that this array has dimension 3*nextatom. I have to check the code.